src/components/flowchart/flowchart.js

```diff
@@ -783,11 +790,17 @@ export class FlowChart extends Component {
           })}
           ref={this.layerNamesRef}
         />
-        {slicedPipelineState.range.length > 0 && (
-          <div ref={this.slicedPipelineActionRef}>
+        {slicedPipeline.length > 0 && (
+          <div ref={this.slicedPipelineActionBarRef}>
             <SlicedPipelineActionBar
               chartSize={chartSize}
-              slicedPipeline={slicedPipelineState.range}
+              displayMetadataPanel={Boolean(clickedNode)}
+              isSlicingPipelineApplied={isSlicingPipelineApplied}
+              onApplySlicingPipeline={() => onApplySlice(true)}
+              onResetSlicingPipeline={this.resetSlicedPipeline}
+              ref={this.slicedPipelineActionBarRef}
```

Comment:
Why do we have duplicate ref like the ref seems to be the same for div and the component. Any reason for this ?

Reply:
ref for div is to create the ref. This `ref={this.slicedPipelineActionBarRef}` is to pass that ref we created from the div to the `SlicedPipelineActionBar` as a props